src/components/hooks/dropdown-hooks.js

```diff
@@ -0,0 +1,67 @@
+import { useRef, useState } from 'react'
+import { useOutsideClick } from 'components/hooks/outside-click'
+
+export const useDropdownHooks = (onChange) => {
+    const [is_open, setOpen] = useState(false)
+    const dropdown_ref = useRef(null)
+    const nodes = new Map()
+
+    useOutsideClick(dropdown_ref, () => setOpen(false))
+
+    const toggleListVisibility = (e) => {
+        const open_dropdown = e.key === 'Space' || e.key === 'Enter' || e.key === 'Tab'
```

Comment:
there are some machines which the key is a number, can we bring keycodes utils back?

Reply:
there was a deprecation notice for the `keyCode` property: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode which why i used `key` instead

Reply:
Tested with different device. Should work as before.

Compatibility is good but currently only IE 11 is not supported. If you want me do make it compatible with IE 11 i could use a mapper to use the newer `key` property with a code mapper as per this SO post (thread no. 3):

https://stackoverflow.com/questions/35394937/keyboardevent-keycode-deprecated-what-does-this-mean-in-practice